src/lib.rs

```diff
@@ -29,62 +40,68 @@ impl <const W: usize, const H: usize> Content<W, H> {
 /// This receives the `SmartLedsWriter` trait implementations along with a
 /// `MatrixType` that describes the size and the pixels mapping between the LED
 /// strip placement and the matrix's x y coordinates.
-pub struct SmartLedMatrix<T, M: MatrixType, const W: usize, const H: usize> {
+pub struct SmartLedMatrix<T, M: Transformation<W, H>, const W: usize, const H: usize> {
     writer: T,
     content: Content<W, H>,
-    matrix_type: M,
+    matrix_type: PhantomData<M>,
     brightness: u8,
 }
 
-impl<T, M: MatrixType, const W: usize, const H: usize> SmartLedMatrix<T, M, W, H> {
+impl<T, M: Transformation<W, H>, const W: usize, const H: usize> SmartLedMatrix<T, M, W, H> {
     pub fn set_brightness(&mut self, new_brightness: u8) {
```

Comment:
It might be useful to also have a getter for the brightness.

Reply:
done